### PR TITLE
[Imgui] Update docking-experimental and freetype features to 1.81

### LINKED ISSUE #16276: imgui[docking-experimental,glfw-binding,opengl3-glbinding-binding] version 1.81#1 build failure
**Host Environment**
- OS: Windows
- Compiler: cl.exe (Visual Studio 2019)

**To Reproduce**
`./vcpkg install imgui[docking-experimental,glfw-binding,opengl3-glbinding-binding]`

**Failure logs** 
vcpkg install imgui[docking-experimental,glfw-binding,opengl3-glbinding-binding]
Computing installation plan...
The following packages will be built and installed:
    imgui[core,docking-experimental,glfw-binding,opengl3-glbinding-binding]:x86-windows -> 1.81#1
Detecting compiler hash for triplet x86-windows...
Could not locate cached archive: C:\Users\Runik\AppData\Local\vcpkg\archives\9c\9c737409ab92c71ab19f461e32c97ac14bc96603.zip
Starting package 1/1: imgui:x86-windows
Building package imgui[core,docking-experimental,glfw-binding,opengl3-glbinding-binding]:x86-windows...
-- Note: imgui only supports static library linkage. Building static library.
-- Using cached D:/Dev/Libraries/vcpkg/downloads/ocornut-imgui-682249396f02b8c21e5ff333ab4a1969c89387ad.tar.gz
-- Cleaning sources at D:/Dev/Libraries/vcpkg/buildtrees/imgui/src/69c89387ad-b99c0b72cb.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Dev/Libraries/vcpkg/downloads/ocornut-imgui-682249396f02b8c21e5ff333ab4a1969c89387ad.tar.gz
-- Using source at D:/Dev/Libraries/vcpkg/buildtrees/imgui/src/69c89387ad-b99c0b72cb.clean
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: ninja -v
    Working Directory: D:/Dev/Libraries/vcpkg/buildtrees/imgui/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\Dev\Libraries\vcpkg\buildtrees\imgui\config-x86-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:319 (vcpkg_execute_required_process)
  ports/imgui/portfile.cmake:64 (vcpkg_configure_cmake)
  scripts/ports.cmake:131 (include)

Error: Building package imgui:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: imgui:x86-windows
  Vcpkg version: 2021-01-13-768d8f95c9e752603d2c5901c7a7c7fbdb08af35

Additionally, attach any relevant sections from the log files above.
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/6002112/config-x86-windows-out.log)

**Additional context**
Tried to copy the missing imgui_tables.cpp to the directory using --editable switch, it fixes this one error but throws a bunch more ... so I reverted back to version 1.79#4 for now.

### PR DESCRIPTION
**Pull request by RT2Code**
Update the docking-experimental feature from 1.79 to 1.81.

The Freetype integration procedure was changed with 1.81, the IMGUI_ENABLE_FREETYPE macro should be defined in imconfig.h.

> - We refactored some of imgui_freetype to make integration easier (a simple #define IMGUI_ENABLE_FREETYPE in your config file does the job for most users) and also to fix some inconsistencies. The majority of imgui_freetyoe users shouldn't be affected as the RasterizerFlags were rarely used.
> - Removed ImGuiFreeType::BuildFontAtlas(). Kept inline redirection function.
> - Prefer using #define IMGUI_ENABLE_FREETYPE, but there's a runtime selection path available too.
> - The shared extra flags parameters (very rarely used) are now stored in ImFontAtlas::FontBuilderFlags.
> - Renamed ImFontConfig::RasterizerFlags (used by FreeType) to ImFontConfig::FontBuilderFlags.
> - Renamed ImGuiFreeType::XXX flags to ImGuiFreeTypeBuilderFlags_XXX for consistency with other API.

I updated the CMake file to uncomment this macro in imconfig.h when the user ask for the freetype feature.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The regressions caused by `gsoap` will be fixed by PR #16270.

File: ports/imgui/CMakeLists.txt [hunk 1]
```diff
@@ -120,6 +120,9 @@ if(IMGUI_FREETYPE)
     find_package(freetype CONFIG REQUIRED)
     target_link_libraries(${PROJECT_NAME} PUBLIC freetype)
     target_sources(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/misc/freetype/imgui_freetype.cpp)
+    FILE(READ ${CMAKE_CURRENT_SOURCE_DIR}/imconfig.h IMCONFIG)
+    STRING(REGEX REPLACE "//#define IMGUI_ENABLE_FREETYPE" "#define IMGUI_ENABLE_FREETYPE" IMCONFIG "${IMCONFIG}")
+    FILE(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/imconfig.h "${IMCONFIG}")
```
Comment (NancyLi1013) [inline ports/imgui/CMakeLists.txt:125]: Could you please use `vcpkg_replace_string()` here?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/imgui/CMakeLists.txt (hunk 1 above)
Comment (JackBoosY) [reply ports/imgui/CMakeLists.txt:125]: @NancyLi1013  No, it's in `CMakelists.txt`, not `portfile.cmake`.

File: ports/imgui/portfile.cmake [hunk 2]
```diff
@@ -4,8 +4,8 @@ if ("docking-experimental" IN_LIST FEATURES)
     vcpkg_from_github(
        OUT_SOURCE_PATH SOURCE_PATH
        REPO ocornut/imgui
-       REF 682249396f02b8c21e5ff333ab4a1969c89387ad
-       SHA512 95f17c14e0a8f10dfc51fd1b30894f9905433fac8f9a93b6c545a542df5eb20be68f40996080a85cba934107ce19fff91a1df1edad1a1b5a0030e8f626e1985d
+       REF 239d09804d17997e147f4bcfb451ead04c1d67ff
```
Comment (JackBoosY) [inline ports/imgui/portfile.cmake:7]: Was the version changed?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/imgui/portfile.cmake (hunk 2 above)
Comment (RT222) [reply ports/imgui/portfile.cmake:7]: Yes. On the current imgui port the library was updated to the 1.81 version, but the docking-experimental feature was left to the 1.79 version (https://github.com/ocornut/imgui/commit/682249396f02b8c21e5ff333ab4a1969c89387ad), so I updated it to the commit corresponding to the 1.81 version (https://github.com/ocornut/imgui/commit/239d09804d17997e147f4bcfb451ead04c1d67ff). Every time the master branch receive a new release version, the docking branch gets it too and also need to be updated in the port.

**Review by vicroms — APPROVED** [review]

File: ports/imgui/CMakeLists.txt (hunk 1 above)
Comment (NancyLi1013) [reply ports/imgui/CMakeLists.txt:125]: Sorry, it's my fault.

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by ras0219-msft — APPROVED** [review]
Editing the sources during configure causes a race condition if we configure release and debug in parallel, so I've moved the edit to the portfile as a post-build step.

As a drive-by, I've changed the versioning scheme to relaxed which will enable our new versioning features.

This LGTM, though I'd appreciate independent confirmation from @RT222 that this still fixes their issue :)

Thanks for the PR!

File: ports/imgui/CMakeLists.txt [hunk 3]
```diff
@@ -120,6 +120,7 @@ if(IMGUI_FREETYPE)
     find_package(freetype CONFIG REQUIRED)
     target_link_libraries(${PROJECT_NAME} PUBLIC freetype)
     target_sources(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/misc/freetype/imgui_freetype.cpp)
+    target_add_definitions(${PROJECT_NAME} PUBLIC IMGUI_ENABLE_FREETYPE)
```
Comment (JackBoosY) [inline ports/imgui/CMakeLists.txt:123]:
```suggestion
    target_compile_definitions(${PROJECT_NAME} PUBLIC IMGUI_ENABLE_FREETYPE)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by RT222** [comment]
The IMGUI_ENABLE_FREETYPE macro should only be defined in imconfig.hpp when the freetype feature is enabled, I fixed it. I also moved the string replacement code for wchar32 to the portfile for the reasons you mentioned about parallel configuration.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**